cmd/relay/relay.go

```diff
@@ -169,3 +155,132 @@ func Run(ctx context.Context, config Config) error {
 		}
 	}
 }
+
+// newENRHandler returns a handler that returns the nodes ID and public address encoded as a ENR.
+func newENRHandler(ctx context.Context, tcpNode host.Host, p2pKey *ecdsa.PrivateKey, config p2p.Config) func(ctx context.Context) ([]byte, error) {
+	// Resolve external hostname peridodically.
+	var (
+		extHostMu sync.Mutex
+		extHostIP net.IP
+	)
+	go func() {
+		if config.ExternalHost == "" {
+			return
+		}
+
+		resolveExtHost := func() {
+			ip, err := net.LookupIP(config.ExternalHost)
+			if err != nil {
+				log.Warn(ctx, "Failed to resolve external host", err, z.Str("host", config.ExternalHost))
+				return
+			}
+			extHostMu.Lock()
+			extHostIP = ip[0]
+			extHostMu.Unlock()
+		}
+
+		onStartup := make(chan struct{}, 1)
+		onStartup <- struct{}{}
+
+		for {
+			select {
+			case <-onStartup:
+				resolveExtHost()
+			case <-time.After(5 * time.Minute):
+				resolveExtHost()
+			case <-ctx.Done():
+				return
+			}
+		}
```

Comment:
```suggestion
		resolveExtHost()

		for {
			select {
			case <-time.After(5 * time.Minute):
				resolveExtHost()
			case <-ctx.Done():
				return
			}
		}
```

Reply:
can we remove the `onStartup` channel like the above?

Reply:
I prefer to have all scheduling in the same for/select loop